.github/CONTRIBUTING.md

```diff
@@ -8,6 +8,16 @@ The following are certain Google Provider-specific things to be aware of when co
 
 We aim to make the Google Provider a good steward of Go practices. See https://github.com/golang/go/wiki/CodeReviewComments for common Go mistakes that you should attempt to avoid.
 
+## Autogenerated Resources
+
+We maintain 2 different versions of the google terraform provider; the [GA (General Audience) provider](https://github.com/terraform-providers/terraform-provider-google) and the [Beta provider](https://github.com/terraform-providers/terraform-provider-google). We are using code generation tool called Magic Modules that uses the same code base to generate both of these providers. Some Terraform resources are defined in `api.yaml` files [located in magic modules](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/products) and are fully generated, whereas some resources are hand written and located in [the provider/terraform/ folder in magic modules](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/provider/terraform/resources). Generated resources will have a prominent header at the top of the file identifying them.
```

Comment:
```suggestion
There are two versions of the Google provider; the [`google` provider](https://github.com/terraform-providers/terraform-provider-google)
and the [`google-beta` provider](https://github.com/terraform-providers/terraform-provider-google-beta).
The `google` provider supports GA ([general availability](https://cloud.google.com/terms/launch-stages))
features, and `google-beta` supports Beta features.

Both providers are generated by our code generation tool, [Magic Modules](https://github.com/GoogleCloudPlatform/magic-modules).
Magic Modules uses a shared code base to generate both providers.
Some Terraform resources are autogenerated. these resources are defined by yaml files called `api.yaml` located in the directory corresponding to their product under the
[Magic Modules `products` directory](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/products).
Autogenerated resources have a prominent header at the top of the file
identifying them.

Handwritten resources have been moved to Magic Modules as well, and are located in
[the `third_party` directory](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform).
Some handwritten resources are the same in both providers; they are stored in
Magic Modules as `.go` files.

Other resources ("compiled" files) are different in each of `google` and `google-beta`;
they'll have the filetype `.go.erb`, and we switch what code is generated based on which
provider is being generated in that run of Magic Modules. 
```

I had some suggestions around reflowing + content, but I could only make a single suggestion because the content was all on one line; feel free to take all or to take out content piecemeal.